### PR TITLE
JENA-1929: Detect and use TDB database type for existing database.

### PR DESCRIPTION
**Pull request by afs**

### CODE REVIEW

File: jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/cmd/DSGSetup.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+package org.apache.jena.fuseki.cmd;
+
+import jena.cmd.CmdException;
+import org.apache.jena.atlas.io.IO;
+import org.apache.jena.fuseki.mgt.Template;
+import org.apache.jena.fuseki.system.spot.TDBOps;
+
+public class DSGSetup {
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/cmd/DSGSetup.java:26]: Would it make sense to combine `cmds/DSGSetup` and this `DSGSetup`? Or maybe add a comment saying that that one is for `ServerConfig`, and this one for `FusekiArgs`?

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB1.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        // Journal, fixed name.
+        validate(location, Names.journalFileBase, Names.extJournal);
+
+        if ( ! isTDB1(location) )
+            throw new TDBException("Not a TDB1 location: "+location);
+
+        // Places for StoreParams: location or default
+        StoreParams params = getStoreParams(location);
+
+        validate(location, Names.journalFileBase, Names.extJournal);
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB1.java:135]: I think this is being validated a few lines above already?

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB2.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        if ( isEmpty(location) )
+            return;
+
+        // Journal, fixed name.
+        validate(location, Names.journalFileBase, Names.extJournal);
+
+        // Places for StoreParams: location or default
+        StoreParams params = getStoreParams(location);
+
+        validate(location, Names.journalFileBase, Names.extJournal);
```
Comment (kinow) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB2.java:161]: Repeated validation? :point_up:

Comment (afs) [reply jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB2.java:161]: Fixed.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB1.java (hunk 2 above)
Comment (afs) [reply jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB1.java:135]: Fixed

File: jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/cmd/DSGSetup.java (hunk 1 above)
Comment (afs) [reply jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/cmd/DSGSetup.java:26]: Good point. renamed one and also made them package scoped.

File: jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/webapp/FusekiWebapp.java [hunk 4]
```diff
@@ -284,13 +287,15 @@ private static DataAccessPoint configFromTemplate(String templateFile, String da
...
-            }
+            if ( ! Objects.equals(dir, Names.memName) && !FileOps.exists(dir) )
+                throw new CmdException("Directory not found: " + dir);
+//            if ( Objects.equals(dir, Names.memName) ) {
+//                Fuseki.configLog.info("TDB dataset: in-memory");
+//            } else {
+//                if ( !FileOps.exists(dir) )
+//                    throw new CmdException("Directory not found: " + dir);
+//                Fuseki.configLog.info("TDB dataset: directory=" + dir);
+//            }
```
[the remark this answers is not in the document]
Comment (afs) [reply jena-fuseki2/jena-fuseki-webapp/src/main/java/org/apache/jena/fuseki/webapp/FusekiWebapp.java:298]: Done!

**Pull request merged**